lib/maildir/unique_name.rb

```diff
@@ -7,7 +7,7 @@ class << self
     def counter
       @counter_mutex ||= Mutex.new
       @counter_mutex.synchronize do
-        @counter = @counter.to_i + 1
+        @counter = (@counter ||=  0) + 1
```

Comment:
there was a an 'uninitialized variable' warning for `@counter` that's what I was going for.  This is one approach to doing that.